backend/app/Models/Listing.php

```diff
@@ -139,6 +161,20 @@ public static function listingsResponse($listings) {
         ];
     }
 
+    public static function listingNotFoundResponse() {
+        return [
+            'success' => false,
+            'error' => 'Listing not found',
+        ];
+    }
+
+    private static function successfulTransactionResponse() {
+        return [
+            'success' => true,
+            'message' => 'Successful transaction.',
+        ];
+    }
+
```

Comment:
@caraencabo 

since you are already using a method to return a response, I suggest that you use `TRAITS` instead.

Traits are basically a collection of reusable methods that are accessible in multiple classes, so in your case, you can create a `ResponseHandlingTrait` and you can apply it to this logic and to your controller responses.

Use this link as reference: https://chasingcode.dev/blog/simple-http-response-trait-laravel/

Reply:
@ragg09-sunasterisk 

This is noted, sir.  I will read and search about it then implement the necessary changes in my next PR.